src/cryptography/x509/base.py

```diff
@@ -1169,5 +1184,69 @@ def build(self, backend: typing.Any = None) -> RevokedCertificate:
         )
 
 
+class Profile(utils.Enum):
+    RFC5280 = 0
+    WebPKI = 1
+
+
+class PolicyBuilder:
+    def __init__(
+        self,
+        *,
+        name: typing.Optional[str] = None,
+        time: typing.Optional[datetime.datetime] = None,
+        profile: Profile = Profile.WebPKI,
+    ):
+        self._name = name
+        self._time = time
+        self._profile = profile
+
+    def name(self, new_name: str) -> PolicyBuilder:
+        """
+        Sets the expected certificate subject.
+        """
+        self._name = new_name
+        return self
+
+    def time(self, new_time: datetime.datetime) -> PolicyBuilder:
+        """
+        Sets the validation time.
+        """
+        self._time = new_time
+        return self
+
+    def profile(self, new_profile: Profile) -> PolicyBuilder:
+        """
+        Sets the underlying profile for this policy.
+        """
+        self._profile = new_profile
+        return self
+
+    def build(self) -> Policy:
+        """
+        Construct a `Policy` from this `PolicyBuilder`.
+        """
+
+        return rust_x509.create_policy(self._profile, self._name, self._time)
+
+
+# NOTE: At the Python level, this is just a thin wrapper around a list
+# of certificates. This is done solely for API misuse prevention reasons:
+# having a dedicated type makes it harder for users to mix up
+# their trusted and untrusted certificate sets.
+class Store:
+    def __init__(self, certs: typing.List[Certificate]):
+        self._certs = certs
+
+
+def verify(
+    leaf: Certificate,
+    policy: rust_x509.Policy,
+    intermediates: typing.List[Certificate],
+    store: Store,
+):
+    return rust_x509.verify(leaf, policy, intermediates, store)
```

Comment:
I wrapped `verify` like this because we can't re-export directly from `rust_x509` (i.e. `from rust_x509 import verify`). Let me know if there's some better/other way to do this.

(I figure this wrapper will also be the right place to make various parameters optional or defaulted, so that we don't have to do it at the Rust layer.)

Reply:
`verify = rust_x509.verify` + `__all__ = ["verify"]` works

Reply:
https://github.com/pyca/cryptography/pull/8873/commits/09f4bdac87f7b14c971ab215ccb75f714abf730f